code/datums/mapgen/_MapGenerator.dm

```diff
@@ -0,0 +1,45 @@
+///All possible biomes in assoc list as type || instance
+var/list/biomes = list()
+
+///Initialize all biomes, assoc as type || instance
+proc/initialize_biomes()
+	for(var/biome_path in concrete_typesof(/datum/biome))
+		var/datum/biome/biome_instance = new biome_path()
+		biomes[biome_path] += biome_instance
+
+///This type is responsible for any map generation behavior that is done in areas, override this to allow for area-specific map generation. This generation is ran by areas in initialize.
```

Comment:
```suggestion
///This type is responsible for any map generation behavior that is done in areas, override this to allow for area-specific map generation. This generation is ran by areas in New.
```

Reply:
Oh I need to fix that comment. I ended up not using New as the biomes weren't initialized in time for the area New proc to run